File: golang-backend/internal/handlers/projecthistoryhandler/dto.go
package projecthistoryhandler

import "github.com/aportela/doneo/internal/handlers/userhandler"

type ProjectHistoryOperationResponse struct {
	ID            string                       `json:"id"`
	CreatedBy     userhandler.UserBaseResponse `json:"createdBy"`
	CreatedAt     int64                        `json:"createdAt"`
	OperationType uint                         `json:"operationType"`
}

type searchResponse struct {
	HistoryOperations []ProjectHistoryOperationResponse `json:"historyOperations"`
}

File: vite-vue-ts-frontend/src/modules/project-history-operations/types/dto.ts
import type { UserBaseResponse } from "../../users/types/dto";

export type ProjectHistoryOperationResponse = {
  id: string;
  createdBy: UserBaseResponse;
  createdAt: number;
  operationType: number;
};

export type SearchResponse = {
  historyOperations: ProjectHistoryOperationResponse[];
};
